source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,311 @@
+for each option.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Node.js Option
+     - Driver Option
+     - Description
+
+   * - **Name**: ``ca``, **Type**: ``Buffer``
+     - | **Name**: ``tlsCAFile``, **Type**: ``string``
+       |
+       | **Name**: ``ca``, **Type**: ``Buffer``
+     - Specifies either one or more certificate authorities to
+       trust when making a TLS connection.
+
+   * - **Name**: ``cert``, **Type**: ``Buffer``
+     - | **Name**: ``tlsCertificateFile``, **Type**: ``string``
+       |
+       | **Name**: ``tlsCertificateKeyFile``, **Type**: ``string``
+       |
+       | **Name**: ``cert``, **Type**: ``Buffer``
+     - Specifies the client certificate file or the client
+       private key file.
+
+   * - **Name**: ``key``, **Type**: ``Buffer``
+     - | **Name**: ``tlsCertificateKeyFile``, **Type**: ``string``
+       |
+       | **Name**: ``key``, **Type**: ``Buffer``
+     - Specifies the client private key file.
+
+   * - **Name**: ``passphrase``, **Type**: ``string``
+     - **Name**: ``tlsCertificateKeyFilePassword``, **Type**: ``string``
+     - Specifies the password to decrypt the client private key.
+
+.. note:: Additional TLS Client Options
+   
+   For **testing purposes**, you can set the ``tlsAllowInvalidHostnames``,
+   ``tlsAllowInvalidCertificates``, and ``tlsInsecure`` client options.
+   
+   Setting the ``tlsAllowInvalidHostnames`` option to ``true`` disables
+   hostname verification, and setting the
+   ``tlsAllowInvalidCertificates`` to ``true`` disables certificate
+   validation. Setting the ``tlsInsecure`` option to ``true`` disables
+   both certificate and hostname validation.
+   
+   Specifying any of these options in a production environment makes
+   your application insecure and potentially
+   vulnerable to expired certificates and to foreign processes posing
+   as valid client instances.
+
+For a full list of client options, see :ref:`node-connection-options`.
+
+.. _node-client-tls-connect:
+
+Reference Certificates in a Client
+----------------------------------
+
+You must reference your certificates in your ``MongoClientOptions`` so
+that the server can validate them before the client connects. You can
+reference your certificates in the following ways:
+
+- Create a ``SecureContext`` object to store certificates *(Recommended)*
+- Provide filepath strings that point to your certificates
+- Create ``Buffer`` objects to store certificates
+
+.. _node-tls-securecontext:
+
+Create a ``SecureContext`` Object to Store Certificates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We recommend that you use the ``secureContext`` option to connect with
+the TLS protocol. By using a ``SecureContext`` object to store your
+certificates, you use the Node.js framework for TLS that is independent of the
+driver. When you use this option, you keep all TLS options in a
+reusable object. This is the most organized way to set up
+TLS for your application.
```

Comment:
S: Seems like we're trying to express the idea that using SecureContext has two specific advantages. To my eye, something like this makes that a little clearer:

```suggestion
We recommend that you use the ``secureContext`` option to configure
your TLS connection. ``SecureContext`` objects are native to Node.js
and let you keep all TLS options in a single reusable object. 
```

always happy to workshop!